### PR TITLE
Add support for custom HealthCheckResultHistoryItem models

### PR DESCRIPTION
**Pull request by JackWH**
As discussed in https://github.com/spatie/laravel-health/pull/16 — this PR implements a `model` config parameter, for use by the `EloquentHealthResultStore` class.

I followed the general conventions used by http://github.com/spatie/laravel-activitylog for setting custom models — hopefully that's along the right lines this time.

This PR is backwards-compatible.

Thanks! 🙌

### CODE REVIEW

File: database/migrations/create_health_tables.php.stub [hunk 1]
```diff
@@ -3,12 +3,13 @@
...
 use Illuminate\Database\Schema\Blueprint;
 use Illuminate\Support\Facades\Schema;
+use Spatie\Health\ResultStores\EloquentHealthResultStore;
 
 return new class extends Migration
 {
     public function up()
     {
-        Schema::create('health_check_result_history_items', function (Blueprint $table) {
+        Schema::create(EloquentHealthResultStore::getHistoryItemInstance()->getTable(), function (Blueprint $table) {
```
Comment (freekmurze) [inline database/migrations/create_health_tables.php.stub:12]: For better readability let's add use a temporary variable `$tableName`

File: src/Exceptions/CouldNotSaveResultsInStore.php [hunk 2]
```diff
@@ -16,4 +17,11 @@ public static function make(ResultStore $store, Exception $exception): self
             previous: $exception,
         );
     }
+    
+    public static function doesNotExtendHealthCheckResultHistoryItem(mixed $invalidValue): self
+    {
+        return new self(
+            "You tried to register an invalid HealthCheckResultHistoryItem model: `{$invalidValue}`. A valid model should extend " . HealthCheckResultHistoryItem::class
```
Comment (freekmurze) [inline src/Exceptions/CouldNotSaveResultsInStore.php:24]: Introduce a variable `$className` to put `HealthCheckResultHistoryItem::class`, so you can use string interpolation in the message.

File: src/ResultStores/EloquentHealthResultStore.php [hunk 3]
```diff
@@ -2,21 +2,44 @@
...
 use Spatie\Health\Models\HealthCheckResultHistoryItem;
 use Spatie\Health\ResultStores\StoredCheckResults\StoredCheckResult;
 use Spatie\Health\ResultStores\StoredCheckResults\StoredCheckResults;
 
 class EloquentHealthResultStore implements ResultStore
 {
+    public static function determineHistoryItemModel(): string
+    {
+        $historyItemModel = config(
+            'health.result_stores.' . EloquentHealthResultStore::class . '.model',
```
Comment (freekmurze) [inline src/ResultStores/EloquentHealthResultStore.php:19]: Set a temporary variable with the value of `HealthCheckResultHistoryItem::class` and use string interpolation

File: src/ResultStores/EloquentHealthResultStore.php [hunk 4]
```diff
@@ -2,21 +2,44 @@
...
+        }
+
+        return $historyItemModel;
+    }
+
+    public static function getHistoryItemInstance(): HealthCheckResultHistoryItem
+    {
+        $historyItemClassName = self::determineHistoryItemModel();
+
+        return new $historyItemClassName();
```
Comment (freekmurze) [inline src/ResultStores/EloquentHealthResultStore.php:34]: Add a docblock here to fix the phpstan failure.

File: src/ResultStores/EloquentHealthResultStore.php [hunk 5]
```diff
@@ -31,12 +54,12 @@ public function save(Collection $checkResults): void
 
     public function latestResults(): ?StoredCheckResults
     {
-        if (! $latestItem = HealthCheckResultHistoryItem::latest()->first()) {
+        if (! $latestItem = (self::determineHistoryItemModel())::latest()->first()) {
```
Comment (freekmurze) [inline src/ResultStores/EloquentHealthResultStore.php:57]: Shouldn't we use `static` here?

**Review by freekmurze — CHANGES_REQUESTED** [review]
Thanks! I think there could be an issue when you want to use multiple stores, but I guess not too many people will do that.

**Review by JackWH — COMMENTED** [review]
Thanks for your feedback 😊 Just made the changes, and PhpStan seems happy now — let me know if you'd like any other tweaks. 👌

**Pull request merged**